pallets/phala-world/src/incubation.rs

```diff
@@ -136,7 +212,63 @@ pub mod pallet {
 			nft_id: NftId,
 		) -> DispatchResult {
 			let sender = ensure_signed(origin)?;
-			// TODO: Move to incubation.rs
+			// Ensure that the collection is an Origin of Shell Collection
+			ensure!(
+				Self::is_origin_of_shell_collection_id(collection_id),
+				Error::<T>::WrongCollectionId
+			);
+			// Ensure that Origin of Shell exists or is not past the hatch time
+			let hatch_time = Self::has_valid_hatch_time(collection_id, nft_id)?;
+			ensure!(!Self::can_hatch(hatch_time), Error::<T>::CannotSendFoodToOriginOfShell);
+			// Check if account owns an Origin of Shell NFT
+			ensure!(
+				pallet_uniques::pallet::Pallet::<T>::owned_in_class(&collection_id, &sender)
+					.count() > 0,
+				Error::<T>::CannotSendFoodToOriginOfShell
+			);
+			// Get Current Era
+			let current_era = pallet_pw_nft_sale::Era::<T>::get();
+			// Get number of times fed this era
+			let num_of_times_fed =
+				OriginOfShellFoodStats::<T>::get((collection_id, nft_id), current_era);
+
+			// Update account FoodInfo if not updated or create new FoodInfo for account's first
+			// feeding
+			if FoodByOwner::<T>::contains_key(&sender) {
+				FoodByOwner::<T>::try_mutate(&sender, |food_info| -> DispatchResult {
```

Comment:
If the key is not in the map, `try_mutate` should return a `Non` in the code. So we can check the existence within the closure to save one read from the storage.